filestore_test.go

```diff
@@ -0,0 +1,269 @@
+		}
+
+		for _, test := range tests {
+			tdbCtx, err := NewContext(nil)
+			require.NoError(t, err)
+			err = NewFile(tdbCtx, arrayURI).Import(test.fileURI, test.mimeType)
+			if test.expectError {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+				assertEqualArraySizeAndFileSize(t, arrayURI, test.fileURI)
+			}
+		}
+	})
+}
+
+func TestFileExport(t *testing.T) {
+	t.Run("ArrayDoesNotExist", func(t *testing.T) {
+		tempDir := t.TempDir()
+
+		tdbCtx, err := NewContext(nil)
+		require.NoError(t, err)
+		err = NewFile(tdbCtx, "file:///array/not/exists").Export(filepath.Join(tempDir, "exported"))
+		require.Error(t, err)
+	})
+
+	t.Run("ArrayDoesNotHaveTheFilestoreSchema", func(t *testing.T) {
+		tempDir := t.TempDir()
+
+		arr, err := newTestArray(t)
+		require.NoError(t, err)
+		tdbCtx, err := NewContext(nil)
+		require.NoError(t, err)
+		uri, err := arr.URI()
+		require.NoError(t, err)
+		err = NewFile(tdbCtx, uri).Export(filepath.Join(tempDir, "exported"))
+		require.Error(t, err)
+	})
+
+	t.Run("ArrayContainsFile", func(t *testing.T) {
+		tempDir := t.TempDir()
+
+		arrayURI := "file://" + filepath.Join(tempDir, "array")
+		fileURI := "testdata/VLDB17_TileDB_Page1.pdf"
+		tdbCtx, err := NewContext(nil)
+		require.NoError(t, err)
+		err = NewFile(tdbCtx, arrayURI).CreateAndImport(fileURI, TILEDB_MIME_PDF)
+		require.NoError(t, err)
+
+		exportURI := filepath.Join(tempDir, "exported")
+		tdbCtx, err = NewContext(nil)
+		require.NoError(t, err)
+		err = NewFile(tdbCtx, arrayURI).Export(exportURI)
+		assert.NoError(t, err)
+		assertSameContents(t, fileURI, exportURI)
+	})
+}
+
+func TestFileReader(t *testing.T) {
+	tempDir := t.TempDir()
+
+	t.Run("ArrayDoesNotExist", func(t *testing.T) {
+		tdbCtx, err := NewContext(nil)
+		require.NoError(t, err)
+		_, err = NewFileReader(tdbCtx, "file:///array/does/not/exist")
+		require.Error(t, err)
+	})
+
+	t.Run("ArrayContainsFile", func(t *testing.T) {
+		// create a filestore array
+		arrayURI := "file://" + filepath.Join(tempDir, "array")
+		fileURI := "testdata/VLDB17_TileDB_Page1.pdf"
+		tdbCtx, err := NewContext(nil)
+		require.NoError(t, err)
+		err = NewFile(tdbCtx, arrayURI).CreateAndImport(fileURI, TILEDB_MIME_PDF)
+		require.NoError(t, err)
+
+		// create a reader
+		tdbCtx, err = NewContext(nil)
+		require.NoError(t, err)
+		r, err := NewFileReader(tdbCtx, arrayURI)
+		require.NoError(t, err)
+		t.Cleanup(func() { r.Close() })
+
+		// export the data via the reader
+		var sink bytes.Buffer
+		n, err := io.Copy(&sink, r)
+
+		// check read cound
+		require.Equal(t, r.Size(), n)
+
+		// verify the exported and imported data are the same
+		fileData, err := os.ReadFile(fileURI)
+		require.NoError(t, err)
+		require.Equal(t, sha1.New().Sum(fileData), sha1.New().Sum(sink.Bytes()))
+	})
+}
+
+func TestFileWriter(t *testing.T) {
+	t.Run("CreateSchema", func(t *testing.T) {
+		tempDir := t.TempDir()
+
+		fileURI := "testdata/VLDB17_TileDB_Page1.pdf"
```

Comment:
test above look much simple with the tables, not sure these ones are as simple or easy to follow